src/nimony/semdecls.nim

```diff
@@ -634,6 +643,24 @@ proc semProc(c: var SemContext; it: var Item; kind: SymKind; pass: PassKind) =
   producesVoid c, info, it.typ
   publish c, symId, declStart
 
+proc semProc(c: var SemContext; it: var Item; kind: SymKind; pass: PassKind) =
+  var n = it.n
+  var typ = asRoutine(n).params
```

Comment:
With a bit of creativity you can move this statement into the uncommon `n.kind == DotToken` case.